fbw-a32nx/src/systems/extras-host/modules/flightplan_sync/FlightPlanAsoboSync.ts

```diff
@@ -17,44 +17,125 @@ export class FlightPlanAsoboSync {
 
     private facilityLoaderCustom: FacilityLoader;
 
+    private cruiseFlightLevel = undefined;
+
+    private originAirport = undefined;
+
+    private destinationAirport = undefined;
+
+    private procedureDetails: ProcedureDetails = undefined;
+
+    private enrouteLegs: (SerializedFlightPlanLeg | Discontinuity)[] = undefined;
+
     constructor(private readonly bus: EventBus) {}
 
     connectedCallback(): void {
-        const sub = this.bus.getSubscriber<FlightPlanSyncEvents>();
+        const sub = this.bus.getSubscriber<FlightPlanSyncEvents & PerformanceDataFlightPlanSyncEvents<A320FlightPlanPerformanceData>>();
 
         this.facilityLoaderCustom = new FacilityLoader(FacilityRepository.getRepository(this.bus));
 
         RegisterViewListener('JS_LISTENER_FLIGHTPLAN', () => {
             this.isReady = true;
         });
-        sub.on('flightPlanManager.syncResponse').handle(async (event) => {
-            await this.syncFlightPlanToGame(event);
-        });
-    }
 
-    public update(): void {
+        // initial sync
         if (NXDataStore.get('FP_SYNC', 'LOAD') === 'SAVE') {
             const pub = this.bus.getPublisher<FlightPlanSyncEvents>();
             pub.pub('flightPlanManager.syncRequest', undefined, true);
         }
+
+        sub.on('flightPlanManager.syncResponse').handle(async (event) => {
+            console.log('SYNC RESPONSE', event);
+            const plan = event.plans[FlightPlanIndex.Active];
+            this.enrouteLegs = plan.segments.enrouteSegment.allLegs;
+            this.originAirport = plan.originAirport;
+            this.destinationAirport = plan.destinationAirport;
+
+            // TODO not really needed anymore
+            this.procedureDetails = {
+                originRunway: plan.originRunway,
+                departureIdent: plan.segments.departureSegment?.procedureIdent,
+                departureTransitionIdent: plan.segments.departureRunwayTransitionSegment?.procedureIdent,
+
+                arrivalIdent: plan.segments.arrivalSegment?.procedureIdent,
+                arrivalTransitionIdent: plan.segments.arrivalEnrouteTransitionSegment?.procedureIdent,
+                arrivalRunwayTransitionIdent: plan.segments?.arrivalRunwayTransitionSegment.procedureIdent,
+                destinationRunway: plan.destinationRunway,
+                approachIdent: plan.segments.approachSegment?.procedureIdent,
+                approachTransitionIdent: plan.segments.approachViaSegment?.procedureIdent,
+            };
+
+            await this.syncFlightPlanToGame();
+        });
+
+        sub.on('flightPlan.setPerformanceData.cruiseFlightLevel').handle(async (event) => {
+            if (event.planIndex === FlightPlanIndex.Active || event.planIndex === FlightPlanIndex.Uplink) {
```

Comment:
We probably don't want to listen to any events regarding the uplink plan, that's meant for temporary plan construction while uplinking from say, simbrief.